modules/core/src/minmax.cpp

```diff
@@ -1510,22 +1509,33 @@ void cv::minMaxIdx(InputArray _src, double* minVal,
 
     Mat src = _src.getMat(), mask = _mask.getMat();
 
-    if (src.dims <= 2)
+    // HAL call
+    if (src.dims <= 2 || src.isContinuous())
     {
-        CALL_HAL(minMaxIdx, cv_hal_minMaxIdx, src.data, src.step, src.cols*cn, src.rows,
-                 src.depth(), minVal, maxVal, minIdx, maxIdx, mask.data);
-    }
-    else if (src.isContinuous())
-    {
-        int res = cv_hal_minMaxIdx(src.data, 0, (int)src.total()*cn, 1, src.depth(),
-                                   minVal, maxVal, minIdx, maxIdx, mask.data);
+        int srcHalStep, srcHalWidth, srcHalHeight;
+        if (src.dims <= 2)
+        {
+            srcHalStep   = src.step;
+            srcHalWidth  = src.cols * cn;
+            srcHalHeight = src.rows;
+        }
+        else // pass it like one continuous row
+        {
+            srcHalStep   = 0;
+            srcHalWidth  = (int)src.total()*cn;
+            srcHalHeight = 1;
+        }
+
+        size_t minOffset = SIZE_MAX, maxOffset = SIZE_MAX;
+        int res = cv_hal_minMaxIdx(src.data, srcHalStep, srcHalWidth, srcHalHeight, src.depth(),
```

Comment:
Why do we need to ignore `CALL_HAL*` macro approach?

Reply:
1. We want HAL function to return offsets instead of indices because HAL function:
 - does not need to know the source matrix shape
 - does not support more than 2 dimensions, so in ND case resulting indices should be fixed anyway
3. Therefore we need to convert offsets to indices
4. `CALL_HAL` macro does not support that, it returns immediately after the call, so it was expanded